craft_parts/ctl.py

```diff
@@ -20,41 +20,65 @@
 import logging
 import os
 import sys
-from typing import List
+from typing import List, Optional
 
 logger = logging.getLogger(__name__)
 
 
-def main():
-    """Run the ctl client cli."""
-    if len(sys.argv) < 2:
-        print(f"usage: {sys.argv[0]} <command> [arguments]")
-        sys.exit(1)
+class CraftCtl:
+    """Client for the craft-parts ctl protocol.
 
-    cmd, param = sys.argv[1], sys.argv[2:]
-    try:
-        client(cmd, param)
-    except RuntimeError as err:
-        logger.error("ctl error: %s", err)
-        sys.exit(1)
+    Craftctl is used to execute built-in step handlers and to get and set
+    variables in the running parts processor context.
+    """
 
+    @classmethod
+    def handle_command(
```

Comment:
this is personal preference, but can we go with the model of registering command handlers instead of using inheritance for this? Any pros and cons you analyzed here?

Reply:
agreed to leave it out for now as nothing needs this

Reply:
Removed from source code.